Sources/KlaviyoSwift/StateManagement.swift

```diff
@@ -0,0 +1,293 @@
+            initialState.initialized = true
+            state = initialState
+            return .run { [state] send in
+                await send(.enqueueRequest(try state.buildProfileRequest()))
+                await send(.start)
+            }
+        case .setEmail(let email):
+            guard state.initialized else {
+                return .none
+            }
+            state.email = email
+            return state.buildProfileTask()
+        case .setAnonymousId(let anonymousId):
+            guard state.initialized else {
+                return .none
+            }
+            state.anonymousId = anonymousId
+            return state.buildProfileTask()
+        case .setPhoneNumber(let phoneNumber):
+            guard state.initialized else {
+                return .none
+            }
+            state.phoneNumber = phoneNumber
+            return state.buildProfileTask()
+        case .setExternalId(let externalId):
+            guard state.initialized else {
+                return .none
+            }
+            state.externalId = externalId
+            return state.buildProfileTask()
+        case .setPushToken(let pushToken):
+            guard state.initialized else {
+                return .none
+            }
+            // TODO: check if we already have this token, skip sending if we do.
+            state.pushToken = pushToken
+            return .task { [state] in
+                return .enqueueRequest(try state.buildTokenRequest())
+            }
+        case .enqueueRequest(let request):
+            guard state.initialized else {
+                return .none
+            }
+            state.queue.append(request)
+            return .none
+        case .flushQueue:
+            guard state.initialized else {
+                return .none
+            }
+            if state.flushing {
+                return .none
+            }
+            if state.queue.isEmpty {
+                return .none
+            }
+            state.requestsInFlight.append(contentsOf: state.queue)
+            state.queue.removeAll()
+            state.flushing = true
+            return .task {
+                .sendRequest
+            }
+        case .stop:
+            guard state.initialized else {
+                return .none
+            }
+            return EffectPublisher.cancel(ids: [RequestId.self, FlushTimer.self])
+                .concatenate(with: .run(operation: { send in
+                    await send(.cancelInFlightRequests)
+                    await send(.archiveCurrentState)
+                }))
+        case .start:
+            guard state.initialized else {
+                return .none
+            }
+            return environment.analytics.timer(state.flushInterval)
+                    .map { _ in
+                        KlaviyoAction.flushQueue
+                    }.eraseToEffect()
+                    .cancellable(id: Timer.self, cancelInFlight: true)
```

Comment:
These cancellables allow us to cancel inflight requests for example when the app goes to the background.